cooperator_website/controllers/main.py

```diff
@@ -0,0 +1,453 @@
+                # contact person values
+                representative = partner.get_representative()
+                values["firstname"] = representative.firstname
+                values["lastname"] = representative.lastname
+                values["gender"] = representative.gender
+                values["email"] = representative.email
+                values["contact_person_function"] = representative.function
+                values["birthdate"] = self.get_date_string(
+                    representative.birthdate_date
+                )
+                values["lang"] = representative.lang
+                values["phone"] = representative.phone
+            else:
+                values["firstname"] = partner.firstname
+                values["lastname"] = partner.lastname
+                values["email"] = partner.email
+                values["gender"] = partner.gender
+                values["birthdate"] = self.get_date_string(partner.birthdate_date)
+                values["lang"] = partner.lang
+                values["phone"] = partner.phone
+        return values
+
+    def fill_values(self, values, is_company, logged, load_from_user=False):
+        sub_req_obj = request.env["subscription.request"]
+        company = request.website.company_id
+        products = self.get_products_share(is_company)
+
+        if load_from_user:
+            values = self.get_values_from_user(values, is_company)
+        if is_company:
+            values["is_company"] = "on"
+        if logged:
+            values["logged"] = "on"
+        values["countries"] = self.get_countries()
+        values["langs"] = self.get_langs()
+        values["products"] = products
+        fields_desc = sub_req_obj.sudo().fields_get(["company_type", "gender"])
+        values["company_types"] = fields_desc["company_type"]["selection"]
+        values["genders"] = fields_desc["gender"]["selection"]
+        values["company"] = company
+
+        if not values.get("share_product_id"):
+            for product in products:
+                if product.default_share_product is True:
+                    values["share_product_id"] = product.id
+                    break
+            if not values.get("share_product_id", False) and products:
+                values["share_product_id"] = products[0].id
+        if not values.get("country_id"):
+            if company.default_country_id:
+                values["country_id"] = company.default_country_id.id
+            else:
+                values["country_id"] = "20"
+        if not values.get("activities_country_id"):
+            if company.default_country_id:
+                values["activities_country_id"] = company.default_country_id.id
+            else:
+                values["activities_country_id"] = "20"
+        if not values.get("lang"):
+            if company.default_lang_id:
+                values["lang"] = company.default_lang_id.code
+
+        comp = request.env["res.company"]._company_default_get()
```

Comment:
This way of fetching the company is deprecated since < 12.0. Plus there is no reason to use two different companies when setting up the `values` dictionnary. Can you use the `company` variable everywhere ?

cf https://github.com/coopiteasy/vertical-cooperative/issues/341